crypto/api/v1/api.go

```diff
@@ -0,0 +1,100 @@
+/*
+ * Nuts crypto
+ * Copyright (C) 2019. Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package v1
+
+import (
+	"errors"
+	"mime"
+	"net/http"
+
+	"github.com/nuts-foundation/nuts-node/crypto"
+	"github.com/nuts-foundation/nuts-node/crypto/log"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
+
+	"github.com/labstack/echo/v4"
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/nuts-foundation/nuts-node/crypto/storage"
+)
+
+// Wrapper implements the generated interface from oapi-codegen
+type Wrapper struct {
+	C crypto.KeyStore
+}
+
+// SignJwt handles api calls for signing a Jwt
+func (w *Wrapper) SignJwt(ctx echo.Context) error {
+	var signRequest = &SignJwtRequest{}
+	err := ctx.Bind(signRequest)
+	if err != nil {
+		log.Logger().Error(err.Error())
+		return echo.NewHTTPError(http.StatusBadRequest, err.Error())
+	}
+
+	if len(signRequest.Kid) == 0 {
+		return echo.NewHTTPError(http.StatusBadRequest, "missing kid")
+	}
+
+	if len(signRequest.Claims) == 0 {
+		return echo.NewHTTPError(http.StatusBadRequest, "missing claims")
+	}
+
+	sig, err := w.C.SignJWT(signRequest.Claims, signRequest.Kid)
+
+	if err != nil {
+		log.Logger().Error(err.Error())
```

Comment:
maybe we can have middleware log errors?

Reply:
I'll add an issue for now